### ISSUE TITLE
sometimes, some processes are not listed

### ISSUE THREAD

**Issue by sam113101** [hishamhm/htop#495]
This is a screenshot of two instances of htop running side by side. One of them lists a firefox process, the other one doesn't. Any idea as to why this happens?

![screenshot](http://i.imgur.com/Sy3IZtm.png)

**Comment by Explorer09** [comment]
It's hard to compare the two lists when you sort the list by "CPU%". The point is, your two instances of htop may not update the process list at the strictly same time. The "CPU%" is measured by ("total cpu time the process is running" - "process cpu time gathered in the last update") / "time interval". So there are always differences when two htop instances do not update the same time.

It seems that the firefox process you mentioned was entering the sleep state, which means it might have no longer been taking CPU time when htop (2nd instance) updates the list. If your htop 1st instance updated it later, it would push firefox further down in the list so you may not notice.

In short: Re-sort the list by PIDs and see if the process is truly missing as you said.

**Comment by sam113101** [comment]
I stared at the two windows for a few minutes. The firefox process never appeared on the left window. I tried sorting the processes by PID as you suggested, and the firefox process was indeed there. I sorted by CPU% again and thereafter it was displayed on top, just like on the other window.

Let me give you more information about the issue. I often keep htop windows running in the background. I rarely restart my computer, so it's not uncommon for me to have htop instances that have been running for a few weeks. Sometimes I notice that some processes are not listed, and restarting htop usually fixes the problem. Now it looks like all I need to do is to change the column it's sorted by.

So, for some reason, the processes are there, but are not sorted properly. Only seems to happen with processes that have been launched *after* htop was launched, newer processes.

**Comment by leofidus** [comment]
I experience the same issue, exactly as described in the previus comment. I often leave htop running for hours, new processes that are started in the meantime often (always?) don't show up newr the top of htop's list even if they should clearly be at the top position. Restarting htop always fixes the issue. Changing the column that htop sorts by seems to also solve the issue (though honestly restarting is faster)

I'm using htop 2.0.2
